src/platform/plugins/shared/visualizations/common/index.ts

```diff
@@ -10,10 +10,180 @@
+  SerializedVisData,
+  SerializedVis,
+  SupportedAggregation,
+  GenericSchemaConfig,
+  SchemaConfig,
+} from './types';
+export type { Dimension, PaletteConfig } from './utils';
+export {
+  prepareLogTable,
+  findAccessor,
+  findAccessorOrFail,
+  getAccessorByDimension,
+  validateAccessor,
+  getColumnByAccessor,
+  isVisDimension,
+  getAccessor,
+  getFormatByAccessor,
+  getStopsWithColorsFromRanges,
+} from './utils';
+export type {
+  Arguments,
+  ExpressionValueVisDimension,
+  ExpressionFunctionVisDimension,
+} from './expression_functions';
+export { range, visDimension } from './expression_functions';
+export type {
+  AggBasedColumn,
+  DataType,
+  TimeScaleUnit,
+  SortingHint,
+  FilterQuery,
+  Filter,
+  Range,
+  NumberValueFormat,
+  MinMax,
+  BasicFullPercentageModeConfig,
+  BasicPercentageModeConfig,
+  PercentageModeConfigWithMinMax,
+  PercentageModeConfig,
+  RangeMode,
+  FormatParams,
+  FiltersParams,
+  TermsParams,
+  DateHistogramParams,
+  RangeParams,
+  MinParams,
+  MaxParams,
+  AvgParams,
+  SumParams,
+  MedianParams,
+  StandardDeviationParams,
+  CardinalityParams,
+  CumulativeSumParams,
+  CounterRateParams,
+  DerivativeParams,
+  CountParams,
+  PercentileParams,
+  PercentileRanksParams,
+  LastValueParams,
+  MovingAverageParams,
+  FormulaParams,
+  StaticValueParams,
+  TimeScaleParams,
+  Operation,
+  OperationWithSourceField,
+  OperationWithReferences,
+  BaseColumn,
+  ColumnWithSourceField,
+  ColumnWithReferences,
+  FiltersColumn,
+  RangeColumn,
+  TermsColumn,
+  DateHistogramColumn,
+  MinColumn,
+  MaxColumn,
+  AvgColumn,
+  SumColumn,
+  MedianColumn,
+  StandardDeviationColumn,
+  CardinalityColumn,
+  PercentileColumn,
+  PercentileRanksColumn,
+  CountColumn,
+  LastValueColumn,
+  CumulativeSumColumn,
+  CounterRateColumn,
+  DerivativeColumn,
+  MovingAverageColumn,
+  FormulaColumn,
+  StaticValueColumn,
+  AnyColumnWithSourceField,
+  AnyColumnWithReferences,
+  Column,
+  GenericColumnWithMeta,
+  ColumnWithMeta,
+  Layer,
+  NavigateToLensContext,
+  ChartShapes,
+  CollapseFunction,
+  FillType,
+  SeriesType,
+  YAxisMode,
+  XYCurveType,
+  PartitionChartType,
+  CategoryDisplayType,
+  NumberDisplayType,
+  LegendDisplayType,
+  LayerType,
+  GaugeColorMode,
+  GaugeShape,
+  GaugeLabelMajorMode,
+  GaugeCentralMajorMode,
+  GaugeTicksPosition,
+  AxisExtentConfig,
+  YConfig,
+  XYDataLayerConfig,
+  XYReferenceLineLayerConfig,
+  EventAnnotationConfig,
+  XYAnnotationsLayerConfig,
+  XYLayerConfig,
+  AxesSettingsConfig,
+  LabelsOrientationConfig,
+  LegendConfig,
+  XYConfiguration,
+  SortingState,
+  PagingState,
+  ColumnState,
+  TableVisConfiguration,
+  MetricVisConfiguration,
+  PartitionLayerState,
+  PartitionVisConfiguration,
+  GaugeVisConfiguration,
+  HeatmapLegendConfig,
+  HeatmapGridConfig,
+  HeatmapArguments,
+  HeatmapLayerState,
+  Palette,
+  HeatmapConfiguration,
+  TagcloudVisConfiguration,
+  Configuration,
+} from './convert_to_lens';
```

Comment:
Yeah I agree with @pmuellr I'm not a huge fan of this rule. If there are _significant_ benefits maybe it's worth it.

I understand it was already in place but have we considered ignoring this rule only for `**/index.ts` files? That's where this hurts the most.

Reply:
I understand the concerns, but I think excluding `**/index.ts` would defeat the purpose.

The rule was created / added by @elastic/kibana-operations with https://github.com/elastic/kibana/pull/109357, and according to the [related issue](https://github.com/elastic/kibana/issues/57370), it's part of our best practices / [styleguide](https://github.com/elastic/kibana/blob/main/STYLEGUIDE.mdx#avoid-export--in-top-level-indexts-files), with the goal of preventing leaking unintended exports.

But I'm open to re-evaluating this decision with @elastic/kibana-operations and perhaps @elastic/kibana-core.